pkg/server/plugin/notifier/k8sbundle/k8sbundle.go

```diff
@@ -155,28 +160,84 @@ func (p *Plugin) getConfig() (*pluginConfig, error) {
 	return p.config, nil
 }
 
-func (p *Plugin) setConfig(config *pluginConfig) {
+func (p *Plugin) setConfig(config *pluginConfig) error {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	p.config = config
+
+	// Start watcher to set CA Bundle in objects created after server has started
+	if p.cancelWatcher != nil {
+		p.cancelWatcher()
+	}
```

Comment:
The order of operations here could be a little more robust. Currently, if there is a failure creating the watcher, 1) the `Configure` call will return an error but will have already mutated the configuration (i.e. `p.config = config`) and 2) the original watcher will have already been shut down. I suggest the following order of events:

- if label configured
    - create new watcher
    - cancel existing watcher and replace `cancelWatcher`
- replace config

Reply:
that makes sense. Only change i would like is to cancel existing watcher outside of ` if label is configured` for the scenario when the label is configures and then the config is updated and the label is no longer configured. We still need to cancel the watcher in that scenario.

Reply:
Yep, good catch.

Reply:
this change is in